client/src/components/exercise/lab8/pages/DataRepair.js

```diff
@@ -0,0 +1,393 @@
+    let error = false;
+    const localMessageError = [...messageError];
+
+    messages.forEach((message, index) => {
+      if (!(message.ai_polarity in repairAllowList)) {
+        // we need to display an error message
+        error = true;
+        localMessageError.splice(index, 1, true);
+      } else {
+        // clear the error message
+        localMessageError.splice(index, 1, false);
+      }
+    });
+    if (!error) {
+      // eventually need to send repair data to the backend
+      setRepairOpen(false);
+      setUserError(false);
+      popUpHandler("The repairs have been made.");
+      setRepairState(true);
+      setIsCorrect(validateCorrectAI());
+    } else {
+      setUserError(true);
+      popUpHandler(
+        "There are errors in your repair. Please correct the errors."
+      );
+    }
+    setMessageError(localMessageError);
+  };
+
+  /**
+   * fetchDataRepair(): is a function responsible for handling retrieval of data to
+   * populate data repair sections.
+   * @returns returns object containing user lab information
+   */
+  const fetchDataRepair = async () => {
+    try {
+      const { userid } = user;
+      return await ExerciseService.getUserRepair(userid);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  /**
+   * postDataRepair(): is a function responsible for handling posting of data to
+   * populate data repair sections.
+   * @returns returns object containing user lab information
+   */
+  const postExerciseChange = async (data) => {
+    try {
+      return await ExerciseService.submitRepair(data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  /**
+   * handleRepair(): is a helper function that is responsible for the
+   * opening and closing of the repair window in this view.
+   */
+  const handleRepair = async () => {
+    !repairOpen ? setRepairOpen(true) : "";
+    const dataRepair = await fetchDataRepair();
+    if (dataRepair?.userid) {
+      const { repair, numRepair, isComplete } = dataRepair;
+      isComplete
+        ? setMessages([...CHAT_MESSAGES.messages])
+        : setMessages([...repair.messages]);
+      setRepairCount(numRepair);
+    }
+  };
+
+  /**
+   * validateCorrectAI(): is a helper function that is responsible for checking
+   * to see if the ai polarity of all of the messages matches the value of the
+   * intended polarity value.
+   */
+  // eslint-disable-next-line no-unused-vars
+  const validateCorrectAI = () => {
+    const localMessages = [...messages];
+    let isCorrectLocal = true;
+    const validateEnteredVsPolarity = (message) => {
+      return message?.ai_polarity === message?.intended_polarity;
+    };
+
+    const messagesOutput = [];
+    localMessages.forEach((message) => {
+      if (validateEnteredVsPolarity(message)) {
+        messagesOutput.push(true);
+      } else {
+        messagesOutput.push(false);
+      }
+    });
+    messagesOutput.forEach((message) => {
```

Comment:
why do we need to iterate over the array again? can't we just set `isCorrectLocal` to false in the first forEach when we push `false` to the messagesOutput array?

Reply:
Good catch